.github/workflows/ci.yml

```diff
@@ -129,39 +129,39 @@ jobs:
     name: Build Windows
     uses: ./.github/workflows/build-native.yml
     with:
-     runner: windows
+     runner: windows-2022
      binary: windows
      build: '["windows"]'
 
   build-macos:
     name: Build MacOS
     uses: ./.github/workflows/build-native.yml
     with:
-     runner: macos
+     runner: macos-11
      binary: macos
      build: '["macos"]'
 
   build-linux:
     name: Build Linux
     uses: ./.github/workflows/build-native.yml
     with:
-     runner: ubuntu
+     runner: ubuntu-20.04
      binary: linux
      build: '["linux"]'
 
   build-android:
     name: Build Android
     uses: ./.github/workflows/build-native.yml
     with:
-     runner: ubuntu
+     runner: ubuntu-20.04
      binary: android
      build: '["android-x86", "android-x86_64", "android-armeabi-v7a", "android-arm64-v8a"]'
 
   build-ios:
     name: Build IOS
     uses: ./.github/workflows/build-native.yml
     with:
-      runner: macos
+      runner: macos-11
```

Comment:
why do we need to do this?

Reply:
There could be the same problem for MacOS as for Linux. We are building with the latest compilator and people with old machine environments may have problem to load the library.

Reply:
Are we fixing an issue or we assume we have an issue with macOS?

Reply:
Yes, it was an assumption. That's why I created https://github.com/realm/realm-dart/pull/1144 to check which environments fail. Since this PR tests don't fail for macos, I will use macos-latest.